pkg/apis/crds/karpenter.k8s.aws_ec2nodeclasses.yaml

```diff
@@ -758,6 +758,13 @@ spec:
                     - Windows2019
                     - Windows2022
                   type: string
+                amiIncludeDeprecated:
```

Comment:
I don't know if we want to add the API surface here. I think we should be able to do this, and it just works:
1. Prioritize AMIs that are not deprecated, prioritizing the newest AMI.
2. If there are no non-deprecated AMIs, use the newest of the deprecated AMIs.

Reply:
@njtran that would mean that the `IncludeDeprecated` field is always set to true on the api call, right? 

By default, if the current account is the AMI owner, all deprecated AMIs appear in the response regardless. If the AMI id specified is a cross account ami that becomes an issue. That AMI is not discoverable at all without this parameter. 

My understanding was that when tags or names are used Karpenter will handle the AMI discovery priortizing the newest AMI always - the issue only comes when pinning using ami ids and if the ami is owned by a different account. 

>Prioritize AMIs that are not deprecated, prioritizing the newest AMI.

This is how it works currently. 
 
>If there are no non-deprecated AMIs, use the newest of the deprecated AMIs.
For this do you think, that if no ami's are returned then try using the `IncludeDeprecated` attribute in the sdk call to describe images?

Reply:
Yeah I think we can just enable it by default in the call, is there a reason not to include it by default? When it comes to priority, Karpenter should currently prioritize the newest AMI regardless if it's tags, names, or ID. We'd just get all the AMIs, both deprecated and not, and filter the results